usr/src/uts/common/fs/zfs/zfs_vnops.c

```diff
@@ -4424,6 +4429,66 @@ zfs_inactive(vnode_t *vp, cred_t *cr, caller_context_t *ct)
 	rw_exit(&zfsvfs->z_teardown_inactive_lock);
 }
 
+static void
+zfs_inactive_task(void *task_arg)
+{
+	znode_t *zp = task_arg;
+	ASSERT(zp);
+	zfs_inactive_impl(zp, CRED());
+}
+
+/*ARGSUSED*/
+void
+zfs_inactive(vnode_t *vp, cred_t *cr, caller_context_t *ct)
+{
+	uint8_t indirection_lvls;
+	dmu_object_info_t doi;
+	znode_t	*zp = VTOZ(vp);
+
+	rw_enter(&zp->z_zfsvfs->z_teardown_inactive_lock, RW_READER);
+	if (zp->z_sa_hdl == NULL) {
```

Comment:
This looks the same as the code in zfs_inactive_impl(), can we break this out into its own function and call it from both places?